source/chemistry/bcl_chemistry_fragment_add_med_chem.cpp

```diff
@@ -240,6 +241,59 @@ namespace bcl
       // mutate label
       BCL_MessageStd( "AddMedChem!");
 
+      // this will cause issues so it's banned
+      if
+      (
+         ( m_TargetMoleculeLinkElementType.empty() && m_EnableDummyAtom ) ||
+         ( !m_TargetMoleculeLinkElementType.empty() && !m_EnableDummyAtom )
+      )
+      {
+        BCL_MessageStd
+        (
+          "\n"
+          "Invalid combination of target molecule atom selection options. "
+          "To obtain pseudo-reaction-style control over the reaction without "
+          "specifying specific atom indices, the following options are available: \n"
+          "1. Set 'mutable_elements=X', do not specify any other atom selectors. \n"
+          "2. Choose an element type that is unique in the target molecule, such as "
+          "Rb, set 'mutable_elements=Rb', 'target_molecule_link_element=Rb', and "
+          "'enable_target_dummy_atom=true'. \n"
+          "For either option 1 or 2, you are free to change the link/dummy element type "
+          "of the medchem fragments via the 'medchem_fragment_link_element' flag depending "
+          "on how your library is constructed."
+          "\n"
+        );
+        return math::MutateResult< FragmentComplete>( util::ShPtr< FragmentComplete>(), *this);
+      }
+      else if
+      (
+          !m_TargetMoleculeLinkElementType.empty() &&
+          m_EnableDummyAtom &&
+          m_MutableElements.Find( GetElementTypes().ElementTypeLookup( m_TargetMoleculeLinkElementType)) >= m_MutableElements.GetSize()
+      )
+      {
+        BCL_MessageStd
+        (
+          "\n"
+          "A custom target molecule link element type was specified and enabled, but "
+          "the input molecule does not contain any elements of the desired type. "
+          "Alternatively, the specified element type is mismatched with the allowed mutable "
+          "element types."
+          "\n"
+        );
+        return math::MutateResult< FragmentComplete>( util::ShPtr< FragmentComplete>(), *this);
+      }
+
+      // get connecting element types
+      const ElementType medchem_fragment_link_element
+      (
+        m_MedChemFragmentLinkElementType.empty() ? GetElementTypes().e_Undefined : GetElementTypes().ElementTypeLookup( m_MedChemFragmentLinkElementType)
+      );
+      const ElementType target_molecule_link_element
+      (
+        m_TargetMoleculeLinkElementType.empty() ? GetElementTypes().e_Undefined : GetElementTypes().ElementTypeLookup( m_TargetMoleculeLinkElementType)
+      );
+
```

Comment:
This logic would also be unnecessary if they are just element types or lists thereof, whichever is appropriate